src/AppInstallerCommonCore/Manifest/ManifestSchemaValidation.cpp

```diff
@@ -98,6 +110,213 @@ namespace AppInstaller::Manifest::YamlParser
 
             return result;
         }
+
+        bool SearchForManifestSchemaHeaderString(std::shared_ptr<std::istream> yamlInputStream, const size_t& rootNodeBeginsAtLine, ManifestSchemaHeader& schemaHeader)
+        {
+            std::string line;
+            size_t currentLine = 1;
+            schemaHeader.SchemaHeaderString.clear();
+
+            // Search for the schema header string in the comments before the root node.
+            while (currentLine < rootNodeBeginsAtLine && std::getline(*yamlInputStream, line))

```

Comment:
This code assumes that the input stream actually contains 8-bit characters.  The YAML parser (wrapper) supports detecting the encoding on the input stream and handling it properly.  The `Parser` type is the one that ends up containing the full string text of the stream, while `Document` contains probably the level that a schema should apply to (we don't really attempt to do anything past the first document in a stream).

Since the YAML parser ignores comments, I would suggest moving the code that detects the schema into the wrapper (probably `Document Parser::Load()`) and making it a property of a `Document`.